stdlib/queue.mli

```diff
@@ -20,6 +20,94 @@
    {b Warning} This module is not thread-safe: each {!Queue.t} value
    must be protected from concurrent access (e.g. with a [Mutex.t]).
    Failure to do so can lead to a crash.
+
+   A basic example:
+    {[
+    # let q = Queue.create ()
+    val q : '_weak1 Queue.t = <abstr>
+
+
+    # Queue.push 1 q; Queue.push 2 q; Queue.push 3 q
+    - : unit = ()
+
+    # Queue.length q
+    - : int = 3
+
+    # Queue.pop q
+    - : int = 1
+
+    # Queue.pop q
+    - : int = 2
+
+    # Queue.pop q
+    - : int = 3
+
+    # Queue.pop q
+    Exception: Stdlib.Queue.Empty.
+    ]}
+
+   For a more elaborate example, a classic algorithmic use of queues
+   is to implement a BFS (breadth-first search) through a graph.
```

Comment:
BFS through a tree would be less involved?

Reply:
I've never really needed a BFS through a tree. I think BFS through a graph (of some kind) is one of the main application of queues, so I illustrate with a real example. Same as above I suppose, it depends on what maintainers think the docs of the stdlib should look like.